app/models/user_mentee_application.rb

```diff
@@ -28,9 +29,60 @@
 #
 class UserMenteeApplication < ApplicationRecord
   belongs_to :user
+  has_many :mentee_application_states, dependent: :destroy
 
   validates :available_hours_per_week, numericality: { greater_than: 0, less_than_or_equal_to: 60 }
 
   validates :city, :state, :country, :reason_for_applying, :learned_to_code, :project_experience,
     :available_hours_per_week, presence: true
+
+  after_create :create_initial_application_state
+
+  def current_status
+    current_state.status
+  end
+
+  def current_state
+    mentee_application_states.last
+  end
+
+  def next_status
+    current_status_index = MenteeApplicationState::STATUSES.keys.index(current_status.to_sym)
+    next_status_index = current_status_index.to_i + 1
+
+    MenteeApplicationState::STATUSES.keys.fetch(next_status_index)
```

Comment:
Outside of the `STATUSES` thing we've already talked about, I think this method is good! One small problem I have is though is the implicit understanding that this method doesn't work and doesn't make sense for `accepted` and `rejected` statuses. It doesn't make sense for the `accepted` status because the calculated next status will be `rejected`, and if you try this method with a `rejected` status, it will error on the `fetch` going out of bounds.

I think it'd be better to have an explicit exception for these two. Maybe have an early return if the current status is accepted or rejected?